### PR TITLE
enh(Resource-status/Actions): Display tooltip for not permitted actions

### PR DESCRIPTION
**Pull request by bdauria**
## Description

This displays a tooltip over deacticated actions when they aren't permitted

Jun-22-2021 14-53-08

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)

## Target serie

- [ ] 20.04.x
- [ ] 20.10.x
- [x] 21.04.x
- [x] 21.10.x (master)

### CODE REVIEW

File: www/front_src/src/Provider/index.tsx [hunk 1]
```diff
@@ -1,3 +1,6 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+
```
Comment (bdauria) [inline www/front_src/src/Provider/index.tsx:3]:
```suggestion
```

File: www/front_src/src/Provider/index.tsx [hunk 2]
```diff
@@ -87,7 +87,7 @@ const AppProvider = (): JSX.Element | null => {
       getUser(userEndpoint),
       getParameters(parametersEndpoint),
       getTranslations(translationEndpoint),
-      getAcl(aclEndpoint),
+      getAcl('http://localhost:5000/mock/acl'),
```
Comment (bdauria) [inline www/front_src/src/Provider/index.tsx:90]:
```suggestion
      getAcl(aclEndpoint),
```

File: www/front_src/src/Provider/index.tsx [hunk 3]
```diff
@@ -137,6 +137,7 @@ const AppProvider = (): JSX.Element | null => {
 
   return (
     <Context.Provider
+      // @ts-ignore
```
Comment (bdauria) [inline www/front_src/src/Provider/index.tsx:140]:
```suggestion
```

**Review by Thebarda — COMMENTED** [review]
The `ActionMenuItem` file is not used here

File: package.json [hunk 4]
```diff
@@ -135,7 +136,7 @@
     "string-argv": "^0.3.1",
     "systemjs": "^6.9.0",
     "systemjs-plugin-css": "^0.1.37",
-    "ts.data.json": "^1.7.0",
+    "ts.data.json": "^2.0.0",
```
[the remark this answers is not in the document]
Comment (bdauria) [reply package.json:139]: It's used directly in centreon though

**Comment by Thebarda** [comment]
There are some "Action not permitted" missing icon button in the resource row
https://www.loom.com/share/f4ff885a71514fbab523b3f74c70301c

**Comment by bdauria** [comment]
> There are some "Action not permitted" missing icon button in the resource row
> https://www.loom.com/share/f4ff885a71514fbab523b3f74c70301c

Fixed

**Review by Jordan91 — APPROVED** [review]
Works well , top :)

**Review by Jordan91 — APPROVED** [review]
Is it normal not to have this behavior on Shortcuts ? 
=> configuration services/hosts etc..

Capture d’écran 2021-06-24 à 13 39 26
Capture d’écran 2021-06-24 à 13 40 07


I think YES , and this issue was groomed during  GROOMING SESSION in this way.

It's not an Action in ACL if i understood well

Capture d’écran 2021-06-24 à 13 45 34

**Review by Thebarda — APPROVED** [review]

**Comment by bdauria** [comment]
> Is it normal not to have this behavior on Shortcuts ?
> => configuration services/hosts etc..
> 
> Capture d’écran 2021-06-24 à 13 39 26
> 
> Capture d’écran 2021-06-24 à 13 40 07
> 
> I think YES , and this issue was groomed during GROOMING SESSION in this way.
> 
> It's not an Action in ACL if i understood well
> 
> Capture d’écran 2021-06-24 à 13 45 34

You are right, shortcuts need the same logic. I think it's better that we implement this behaviour while we move the icons to the header

**Pull request merged**